src/Microsoft.OData.Core/Json/ODataUtf8JsonWriter.cs

```diff
@@ -0,0 +1,266 @@
+            }
+            else
+            {
+                this.writer.WriteNumberValue(value);
+            }
+            
+            this.FlushIfBufferThresholdReached();
+        }
+
+        public void WriteValue(double value)
+        {
+            this.writer.WriteNumberValue(value);
+            this.FlushIfBufferThresholdReached();
+        }
+
+        public void WriteValue(Guid value)
+        {
+            this.writer.WriteStringValue(value);
+            this.FlushIfBufferThresholdReached();
+        }
+
+        public void WriteValue(decimal value)
+        {
+            if (this.isIeee754Compatible)
+            {
+                this.writer.WriteStringValue(value.ToString(CultureInfo.InvariantCulture));
+            }
+            else
+            {
+                this.writer.WriteNumberValue(value);
+            }
+
+            this.FlushIfBufferThresholdReached();
+        }
+
+        public void WriteValue(DateTimeOffset value)
+        {
+            this.writer.WriteStringValue(value);
+            this.FlushIfBufferThresholdReached();
+        }
+
+        public void WriteValue(TimeSpan value)
+        {
+            string stringValue = EdmValueWriter.DurationAsXml(value);
+            this.writer.WriteStringValue(stringValue);
+            this.FlushIfBufferThresholdReached();
+        }
+
+        public void WriteValue(Date value)
+        {
+            this.writer.WriteStringValue(value.ToString());
+            this.FlushIfBufferThresholdReached();
+        }
+
+        public void WriteValue(TimeOfDay value)
+        {
+            this.writer.WriteStringValue(value.ToString());
+        }
+
+        public void WriteValue(byte value)
+        {
+            this.writer.WriteNumberValue(value);
+            this.FlushIfBufferThresholdReached();
+        }
+
+        public void WriteValue(sbyte value)
+        {
+            this.writer.WriteNumberValue(value);
+            this.FlushIfBufferThresholdReached();
+        }
+
+        public void WriteValue(string value)
+        {
+            if (value == null)
+            {
+                this.writer.WriteNullValue();
+            }
+            else
+            {
+                this.writer.WriteStringValue(value);
+            }
+
+            this.FlushIfBufferThresholdReached();
+        }
+
+        public void WriteValue(byte[] value)
+        {
+            this.writer.WriteBase64StringValue(value);
+            this.FlushIfBufferThresholdReached();
+        }
+
+        public void WriteRawValue(string rawValue)
+        {
+            // TODO: Utf8JsonWriter.WriteRawValue supported from .NET 6
+            // fallback to write to buffer directly as UTF8-encoded bytes
+
+            this.Flush(); // ensure we don't write to the stream while there are still pending data in the buffer
+            this.outputStream.Write(encoding.GetBytes(rawValue));

```

Comment:
```suggestion
#if NET6_0
            this.writer.WriteRawValue(rawValue);
#else
            this.Flush(); // ensure we don't write to the stream while there are still pending data in the buffer
            this.outputStream.Write(encoding.GetBytes(rawValue));
#endif
```

Reply:
I don't think our existing CI build will recognize this symbol. I suspect a .NET 6 service will still end up calling the `outputSream.Write` code unless we target .net6.0 and/or use .NET 6 SDK to run the build.

Reply:
I created the issue here: https://github.com/OData/odata.net/issues/2420